### PR TITLE
refactor stdin readline

### PR DESCRIPTION
**Pull request by robatipoor**

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use common::errors::StringErrorResult;
 use gumdrop::Options;
 use server;
 use std::env::current_exe;
-use std::io::{self, Read};
+use std::io::{self, BufRead};
```
Comment (yaa110) [inline cli/src/main.rs:13]: how does it help to use an internal buffer to read the whole content of stdin?

File: common/src/macros/oops.rs [hunk 2]
```diff
@@ -7,8 +7,8 @@
 /// ```
 #[macro_export]
 macro_rules! oops {
-    ($($arg:tt)*) => {
+    ($($arg:tt)*) => {{
```
Comment (yaa110) [inline common/src/macros/oops.rs:10]: this change should be done in another commit

**Review by yaa110 — CHANGES_REQUESTED** [review]

File: cli/src/main.rs (hunk 1 above)
Comment (robatipoor) [reply cli/src/main.rs:13]: iterator over the lines of an instance of BufRead we need this trait is in scope

File: common/src/macros/oops.rs (hunk 2 above)
Comment (robatipoor) [reply common/src/macros/oops.rs:10]: i close pull request and send again please review
